### ISSUE TITLE
Upgrade Hyper example to support Tokio 1.x

### ISSUE THREAD

**Issue by nadenf** [DataDog/glommio#389]
The existing [hyper example](https://github.com/DataDog/glommio/blob/master/examples/hyper.rs) uses an old version of Tokio and by extension Hyper.

There has been a breaking change in Tokio where poll_read now passes in a ReadBuf instead of a byte buffer.

The following code was tried:
```
    impl tokio::io::AsyncRead for HyperStream {
        fn poll_read(
            mut self: Pin<&mut Self>,
            cx: &mut Context,
            buf: &mut ReadBuf<'_>,
        ) -> Poll<io::Result<()>> {
            Pin::new(&mut self.0).poll_read(cx, buf.initialize_unfilled());
            buf.advance(buf.remaining());
            Poll::Ready(Ok(()))
        }
    }
```

But during testing i.e. curl http://127.0.0.1:8000/hello it returns "world" but internally is generating a 400 Bad Request error and prints out a "Transport endpoint is not connected" error message.

**Comment by glommer** [comment]
Thanks @nadenf. I have to read more on the new initialization rules on Tokio buffers to make an intelligent comment. But there are a few things that just look wrong from my bird's eye view on this code.

1. What if the first `poll_read( )` returns `Pending`? You seem to be always returning `Ready` from this code, which doesn't seem right. Likely we have to wrap this in the `ready!()` macro at that point.
2. Similarly, it is strange that you are always doing something with the buffer regardless of what happens inside the call to `poll_read`

**Comment by ChillFish8** [comment]
So the code above is invalid, you need to respect the original poll's response as this is what actually indicates if the resource is ready or not which current is always Ready leaving the state machine to fall into all kinds of madness.

Similarly, you have to advance the buffer by n amount of bytes read which is returned from the Ready(usize) of the poll_read();

So it should become something like this:  (note the bellow code isn't tested as I'm currently just on my windows system.
```rust
    impl tokio::io::AsyncRead for HyperStream {
        fn poll_read(
            mut self: Pin<&mut Self>,
            cx: &mut Context,
            buf: &mut ReadBuf<'_>,
        ) -> Poll<io::Result<()>> {
            Pin::new(&mut self.0).poll_read(cx, buf.initialize_unfilled())?
                .map(|n| {
                    buf.advance(n);
                    Ok(())
                })
        }
    }
```

**Comment by nadenf** [comment]
@ChillFish8  .. Thanks for your guidance. Have fixed it in #388 .
